pkg/sign/notary.go

```diff
@@ -187,63 +233,63 @@ func (ns NotarySigner) Sign(images []string) error {
 	defer w.Stop()
 	for retries > 0 {
 		fmt.Printf("Trying to sign %s. %v retries remaining...\n", sImg, retries)
-		// wait for ticker to run
 		<-w.C
-		resp, err := ns.c.Do(req)
+
+		// Send the HTTP request
+		resp, err := client.Do(req)
 		if err != nil {
-			return fmt.Errorf("notary request: %w", err)
+			return fmt.Errorf("failed to send request: %w", err)
 		}
-		status = resp.Status
+		defer resp.Body.Close()
+
+		// Read the response body
 		respMsg, err = io.ReadAll(resp.Body)
 		if err != nil {
-			return fmt.Errorf("body read: %w", err)
+			return fmt.Errorf("failed to read response: %w", err)
 		}
 		switch resp.StatusCode {
-		case http.StatusOK:
-			// response was fine. Do not need anything else
+		case http.StatusAccepted:
 			fmt.Printf("Successfully signed images %s!\n", sImg)
 			return nil
 		case http.StatusUnauthorized, http.StatusForbidden, http.StatusBadRequest, http.StatusUnsupportedMediaType:
 			return ErrBadResponse{status: status, message: string(respMsg)}
 		}
 		retries--
 	}
-	fmt.Println("Reached all retries. Stopping.")
-	return ErrBadResponse{status: status, message: string(respMsg)}
+
+	return fmt.Errorf("failed to sign images: %s", string(respMsg))
 }
 
 func (nc NotaryConfig) NewSigner() (Signer, error) {
 	var ns NotarySigner
 
-	// (@Ressetkk): Should this be loaded before calling signer?
 	if nc.Secret != nil {
 		f, err := os.ReadFile(nc.Secret.Path)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to read secret file: %w", err)
 		}
+
 		switch nc.Secret.Type {
-		case "token":
-			ns.authFunc = AuthToken(string(f))
 		case "signify":
 			var s SignifySecret
-			err := yaml.Unmarshal(f, &s)
+			err := json.Unmarshal(f, &s)
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("failed to unmarshal signify secret: %w", err)
 			}
-			auth, err := SignifyAuth(s)
-			if err != nil {
-				return nil, err
-			}
-			ns.authFunc = auth
+
+			ns.signifySecret = s
+
 		default:
-			return nil, ErrAuthServiceNotSupported{Service: nc.Secret.Type}
+			return nil, fmt.Errorf("unsupported secret type: %s", nc.Secret.Type)
 		}
 	}
+
 	ns.retryTimeout = 10 * time.Second
 	if nc.RetryTimeout > 0 {
 		ns.retryTimeout = nc.RetryTimeout
 	}
-	ns.c.Timeout = nc.Timeout
```

Comment:
Why this is not needed?

Reply:
removed by mistake, restored

Reply:
You do. It is part of initialisation of HTTP Client few lines above.